### ISSUE TITLE
Regression in 1.16.22: @EqualsAndHashCode includes transient fields by default

### ISSUE THREAD

**Issue by emlun** [rzwitserloot/lombok#1773]
With this code:

```
@EqualsAndHashCode
public class BugExample {

    int a;
    transient int b;

    public BugExample(int a, int b) {
        this.a = a;
        this.b = b;
    }

}
```

the field `b` in fact does make a difference for `.equals`. The regression was introduced in version 1.16.22; 1.16.20 works as documented.

Minimal example with tests exposing the issue: https://github.com/emlun/lombok-1.16.22-regression

Probably related to #1772.

**Comment by Maaartinus** [comment]
I'd bet this has been fixed in 1.18, just can't see it in the [changelog](https://projectlombok.org/changelog).

**Comment by emlun** [comment]
Sorry, yep, it's fixed in f241ef8c309b90faacabe66431b9deaebfd7087a on master. Also duplicate of #1724.

**Comment by emlun** [comment]
I mean it's not fixed in the 1.18.0 release, but I guess the fix will be included in the next release.
